src/fixed.c

```diff
@@ -9,42 +9,66 @@
 #define uint32_lo(i) ((i) & 0xffff)
 #define uint32_hi(i) ((i) >> 16)
 #define uint32_carry16 ((1) << 16)
+/*if (x > minx && x < minx + size) ...*/
+#define RANGE_CHECK(x, minx, size) \
+    ((int32_t) ((x - minx) | (minx + size - 1 - x)) > 0)
 
 twin_fixed_t twin_fixed_sqrt(twin_fixed_t a)
 {
-    twin_fixed_t max = a, min = 0;
-
-    while (max > min) {
-        twin_fixed_t mid = (max + min) >> 1;
-        if (mid >= 181 * TWIN_FIXED_ONE) {
-            max = mid - 1;
-            continue;
-        }
-        twin_fixed_t sqr = twin_fixed_mul(mid, mid);
-        if (sqr == a)
-            return mid;
-        if (sqr < a)
-            min = mid + 1;
-        else
-            max = mid - 1;
+    if (a <= 0)
+        return 0;
+    
+    if (RANGE_CHECK(a, TWIN_FIXED_ONE - (1 << (8 - 1)), (1 << 8)))
```

Comment:
It looks ugly. Consider the following macro:
```
#define CHECK_INTERVAL(x, epsilon) ...
```

Here, the epsilon is exactly `1 << (8 - 1)`.